pkg/locker/locker.go

```diff
@@ -90,6 +93,13 @@ func (l *Locker) Lock(name string) {
 	// once locked then we can decrement the number of waiters for this lock
 	nameLock.Lock()
 	nameLock.dec()
+
+	if nameLock.err != nil {
```

Comment:
If you perform the error check before the Lock() as well, then its an optimization for cases where goroutine1 has already set the error before goroutine2 begins to take the lock. 

What you have here is fine too.